unstructured/partition/email.py

```diff
@@ -1,20 +1,43 @@
 import email
 import sys
+from email.message import Message
 from typing import Dict, IO, List, Optional
 
 if sys.version_info < (3, 8):
     from typing_extensions import Final
 else:
     from typing import Final
 
-from unstructured.cleaners.core import replace_mime_encodings
+from unstructured.cleaners.core import replace_mime_encodings, clean_extra_whitespace
 from unstructured.documents.elements import Element, Text
 from unstructured.partition.html import partition_html
 
 
 VALID_CONTENT_SOURCES: Final[List[str]] = ["text/html"]
 
 
+def partition_attachment_info(message: Message, output_dir: str = None) -> Dict[str, str]:
```

Comment:
I'd suggest renaming this to `extract_attachment_info` since we're saving them off to a directory. Once the attachments are in the directory, you can call the appropriate `partition_*` function to do the partitioning.

Reply:
😂 I changed it from `extract_attachment_info`, but yes I can change the name.

Reply:
🤣 - it's a great name!